synapse/storage/databases/main/schema/delta/60/federation_stream_position_unique_idx.sql

```diff
@@ -0,0 +1,4 @@
+-- Ensure that only one position can be tracked for each stream type.  This might
+-- be violated if, for exapmle, a backup is restored without first clearing the
+-- table's contents.
+CREATE INDEX federation_stream_position_idx ON federation_stream_position(type);
```

Comment:
We should also clear out any duplicates that exist, otherwise this will fail. We should be able to add a delete before the index creation that deletes duplicate rows (we want to keep the row with the min stream position).